Ensure project -I flags are listed before user-specified CFLAGS and
CPPFLAGS which might contain -I flags.

Also set an ARCHFLAGS variable which is used by the Makefile patches.
--- configure.orig	2019-02-07 13:53:38.000000000 -0600
+++ configure	2022-04-18 01:48:39.000000000 -0500
@@ -315,8 +315,10 @@
 
 echo LDFLAGS=$LDFLAGS >> Makefile.settings
 
-echo CFLAGS=$CFLAGS $CPPFLAGS >> Makefile.settings
-echo CFLAGS+=-I${srcdir} -I${srcdir}/lib -I${srcdir}/protocols -I. >> Makefile.settings
+echo CFLAGS=-I${srcdir} -I${srcdir}/lib -I${srcdir}/protocols -I. >> Makefile.settings
+# Potential -I flags in $CPPFLAGS need to come after the above
+echo CFLAGS+=$CFLAGS $CPPFLAGS >> Makefile.settings
+echo ARCHFLAGS=$ARCHFLAGS >> Makefile.settings
 
 echo CFLAGS+=-DHAVE_CONFIG_H -D_GNU_SOURCE >> Makefile.settings
 
